.github/workflows/build.yml

```diff
@@ -0,0 +1,37 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```

Comment:
does this mean it will only run on the master branch? and if so, wouldn't that be before the merge?

Reply:
This means that this will run in every pull request to master end after the merge to master. The last one can be used for creating a release based on the builds artifacts.